lib/Support/Check.cpp

```diff
@@ -60,40 +82,54 @@ static bool testBufferExact(offloadtest::Buffer *B1, offloadtest::Buffer *B2) {
   return true;
 }
 
+template <typename T>
+static bool testAll(std::function<bool(const T &, const T &)> Fn,
```

Comment:
Should we give this parameter a more descriptive name like "ComparisonFn" instead of "Fn" ?